extensions/README.md

```diff
@@ -0,0 +1,152 @@
+new command with blade by using standard JRE service loader mechanisms.  Blade looks for custom commands using
+the this service interfaces: `com.liferay.blade.cli.command.BaseCommand`
+
+Therefore in your extension jar you should have a file like this: `META-INF/services/com.liferay.blade.cli.command.BaseCommand1` and inside would be a list of all of your FQDN classes of your custom commands.
+
+An example of how this should be laid out can be found in the blade repository.
+
+https://github.com/liferay/liferay-blade-cli/tree/master/extensions/sample-command
+
+Pay close attention to a few things in this project
+
+1. Registering your command with blade: https://github.com/liferay/liferay-blade-cli/blob/master/extensions/sample-command/src/main/resources/META-INF/services/com.liferay.blade.cli.command.BaseCommand
+2. You need to build your extension against the Blade CLI API (which is not yet officially published, but you can use the provionsal repo as shown here)
+https://github.com/gamerson/blade-sample-command/blob/master/build.gradle#L18
+3. You can author your tests for your custom commands as well following this example: https://github.com/liferay/liferay-blade-cli/blob/master/extensions/sample-command/src/test/java/com/liferay/extensions/sample/command/CommandsTest.java
+
+Once you have your command, build it with gradle `gradle build` and then you should have your jar file.  Now you can test
+it in blade by running `$ blade extension install /path/to/your/blade-command/build/libs/blade-command.jar` or by simply copying
+the file to ${user.home}/.blade/extensions
+
+** Give it a try! **
+```
+$ blade help
+```
+Now you should see your new custom command listed!.  If not, contact us on our #blade forum on the Liferay community slack channel.  https://community.liferay.com/it/chat
+
+
+### Custom project templates
+
+Blade comes with over 37+ project templates (whew!) right now, but many times you may feel that those are too simple or
+don't fit the right need for your development team.  You can create new custom project templates that fit exactly in
+your own team's workflow and have blade use it instead.  Tto do that simply follow the example receipt here for how to
+build your own custom project template.
+
+https://github.com/liferay/liferay-blade-cli/tree/master/extensions/sample-template
+
+Here are a few things to notice from this sample.
+
+1. This template is structured as a Maven archetype.  Read more about Maven Archetypes and their features and capabilities here: https://maven.apache.org/guides/introduction/introduction-to-archetypes.html
+2. The BundleSymbolic name of your template jar (see here: https://github.com/liferay/liferay-blade-cli/blob/master/extensions/sample-template/bnd.bnd#L3) must have the pattern `*.project.templates.<name>.*`  So lets say you have a custom template called "mywebapp", then the BSN should be `org.myorg.project.templates.mywebapp-1.0.0.jar`
```

Comment:
Perhaps rephrase "BundleSymbolic name" as "Bundle-SymbolicName" to match the name of the parameter, possibly also putting "(BSN)" after so the reader understands the abbreviation.